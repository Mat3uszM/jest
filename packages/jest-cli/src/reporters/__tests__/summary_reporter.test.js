/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
'use strict';

import SummaryReporter from '../summary_reporter';

const env = Object.assign({}, process.env);
const write = process.stderr.write;
const globalConfig = {
  watch: false,
};
const aggregatedResults = {
  numTotalTestSuites: 1,
  snapshot: {
    filesUnmatched: 1,
    unmatched: 2,
  },
  testResults: {},
};

let results = [];

beforeEach(() => {
  process.env.npm_lifecycle_event = 'test';
  process.env.npm_lifecycle_script = 'jest';
  process.stderr.write = result => results.push(result);
});

afterEach(() => {
  results = [];
  process.env = env;
  process.stderr.write = write;
});

test('npm test', () => {
  process.env.npm_config_user_agent = 'npm';
  process.stderr.write = result => results.push(result);
  const testReporter = new SummaryReporter(globalConfig);
  testReporter.onRunComplete(new Set(), aggregatedResults);
  expect(results[0] + results[1]).toMatchSnapshot();
});

test('yarn test', () => {
  process.env.npm_config_user_agent = 'yarn';
  const testReporter = new SummaryReporter(globalConfig);
  testReporter.onRunComplete(new Set(), aggregatedResults);
  expect(results[0] + results[1]).toMatchSnapshot();
});
