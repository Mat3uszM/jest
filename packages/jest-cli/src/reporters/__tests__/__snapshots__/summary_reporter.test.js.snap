// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots needs update with npm test 1`] = `
"Snapshot Summary
 › 2 snapshot tests failed in 1 test suite. Inspect your code changes or run \`npm test -- -u\` to update them.

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   2 failed, 2 total
Time:        0.01s
Ran all test suites.
"
`;

exports[`snapshots needs update with yarn test 1`] = `
"Snapshot Summary
 › 2 snapshot tests failed in 1 test suite. Inspect your code changes or run \`yarn test -u\` to update them.

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   2 failed, 2 total
Time:        0.01s
Ran all test suites.
"
`;
